### DISCUSSION TITLE
Projetos feitos com NextJS

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#891] [Discussão]
Pessoal, quero conhecer projetos que foram feitos com nextjs, se você trabalhou em algum e usou nextjs envia a url, ou se for open source manda o link do source, vale falar dos perrengues que vcs passaram e se ficaram satisfeitos :)

edit
Ele foi mencionado nessa thread que diz respeito a SEO #631

**Comment by felipefialho** [comment]
@fdaciuk

**Comment by fdaciuk** [comment]
Estou trabalhando em alguns projetos com Next. Nenhum está em produção ainda, mas ele resolve bem para o que ele se propõe.

Não precisa de configuração para começar a usar, mas é possível extender a configuração padrão.

Para CSS, só rola com CSS in JS. Ele já disponibiliza uma opção, mas eu estou usando `styled-components` e está resolvendo bem pra mim.

Perdi umas poucas horas pra deixar tudo rodando certinho, mas a configuração básica que eu preciso pra começar qualquer projeto eu deixei nesse repo:
https://github.com/fdaciuk/workflow-nextjs

Ainda preciso documentar e adicionar mais algumas configs, mas vou manter esse repo atualizado, se alguém precisar =)

**Comment by alanhr** [comment]
Eu estou usando Next em um projeto e estou gostando bastante ele resolve bem o problema de SSR

**Comment by fernahh** [comment]
Eu fiz a página web do changelog meetup, é bizarramente simples, o código está aqui: https://github.com/changelogmeetup/website

O que me incomodou um pouco, (mas era antes da vesão 4, talvez já tenham melhorado), era o prefetch de conteúdo. Pra ter uma transição "like spa", tu precisa fazer isso, só que se tu tiver uma lista de 10, 15 links ele vai baixar todo o conteúdo mesmo se o usuário não acessar. Porém, se tu não fizer isso, a transição entre páginas é bem lenta, o que acaba ficando inútil usar JavaScript pra load de conteúdo e mesmo assim ter uma sensação de algo single-server-rendering.

**Comment by felquis** [comment]
@fernahh demorou um bom tempo o first load do https://changelog.now.sh, creio que é uma boa prática fazer o export da versão estática, e por em uma boa CDN a pasta do output.

**Comment by fernahh** [comment]
> @felquis demorou um bom tempo o first load do https://changelog.now.sh, creio que é uma boa prática fazer o export da versão estática, e por em uma boa CDN a pasta do output.

Sim. Na verdade tá num server free, ainda não tirei tempo pra arrumar isso. :sweat_smile:
